File: data/derived_cot/rq3_prompting/results/translation/r1/py_to_cpp/code_output/AvgPartition.cpp
#include <vector>
#include <utility>
#include <algorithm>

class AvgPartition {
public:
    AvgPartition(const std::vector<int>& lst, int limit)
        : lst(lst), limit(limit) {}

    std::pair<int, int> setNum() const {
        int size = lst.size() / limit;
        int remainder = lst.size() % limit;
        return {size, remainder};
    }

    std::vector<int> get(int index) const {
        auto [size, remainder] = setNum();
        int start = index * size + std::min(index, remainder);
        int end = start + size;
        if (index + 1 <= remainder) {
            end += 1;
        }
        return std::vector<int>(lst.begin() + start, lst.begin() + end);
    }

private:
    std::vector<int> lst;
    int limit;
};

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/cpp_to_py/AvgPartition.py
class AvgPartition:

    def __init__(self, lst, limit):
        self.lst = lst
        self.limit = limit

    def setNum(self):
        n = len(self.lst)
        size = n // self.limit
        remainder = n % self.limit
        return (size, remainder)

    def get(self, index):
        size, remainder = self.setNum()
        start = index * size + min(index, remainder)
        end = start + size
        if index < remainder:
            end += 1
        return self.lst[start:end]